### PR TITLE
tp-qemu: format_disk: Improve the debug log.

### PR DESCRIPTION
**Pull request by CongSmile**
1. Improve the debug log.
2. Update the single character.

Signed-off-by: Cong Li <coli@redhat.com>

### CODE REVIEW

**Comment by CongSmile** [comment]
id: 1181930

**Comment by xutian** [comment]
@CongSmile 
CI Failed, Please reformat you patch please!

**Comment by CongSmile** [comment]
Hi @xutian 

I have updated the code, could you help review it again?

Thanks.

File: qemu/tests/cfg/format_disk.cfg [hunk 1]
```diff
@@ -13,7 +13,9 @@
     format_cmd = "cd /dev && for i in `ls | egrep [shv]db`;do yes |mkfs.ext3 $i; done"
     list_disk_cmd = ""
     set_online_cmd = ""
+    show_dev_cmd = "cd /dev && ls | egrep [shv]db"
     mount_cmd =  "mkdir -p /media && cd /dev && ls | egrep [shv]db | xargs -I dev mount -t ext3 dev /media"
+    show_mount_cmd = "mount | grep -Eo '/dev/[shv]db'"
```
Comment (ypu) [inline qemu/tests/cfg/format_disk.cfg]: As this case should support Windows guest, please provide related command for windows guest too.

Comment (CongSmile) [inline qemu/tests/cfg/format_disk.cfg]: Hi @ypu 

I have added debug info when mount / umount failed.
Windows wont execute it.

Maybe you mean add such debug info in other places, could you tell me more details?

Thanks a lot.

File: qemu/tests/format_disk.py [hunk 2]
```diff
@@ -36,42 +36,45 @@ def run(test, params, env):
...
+    show_dev_cmd = params.get("show_dev_cmd")
     if mount_cmd:
         error.context("Mount the disk with cmd '%s'" % mount_cmd, logging.info)
-        s, o = session.cmd_status_output(mount_cmd, timeout=cmd_timeout)
-        if s != 0:
-            raise error.TestFail("Failed to mount with error: %s" % o)
-        logging.info("Output of mount disk command: %s" % o)
+        status, output = session.cmd_status_output(mount_cmd, timeout=cmd_timeout)
+        if status != 0:
+            device_list = session.cmd_output_safe(show_dev_cmd)
```
Comment (xutian) [inline qemu/tests/format_disk.py]: show_dev_cmd  only used here, so  it's better to use params[show_dev_cmd]

File: qemu/tests/format_disk.py [hunk 3]
```diff
@@ -80,31 +83,37 @@ def run(test, params, env):
...
+        status, output = session.cmd_status_output(readfile_cmd, timeout=cmd_timeout)
+        if status != 0:
+            raise error.TestFail("Read file error: %s" % output)
+        if output.strip() != ranstr:
             raise error.TestFail("The content written to file has changed, "
-                                 "from: %s, to: %s" % (ranstr, o.strip()))
+                                 "from: %s, to: %s" % (ranstr, output.strip()))
 
     umount_cmd = params.get("umount_cmd")
+    show_mount_cmd = params.get("show_mount_cmd")
```
Comment (xutian) [inline qemu/tests/format_disk.py]: same as show_dev_cmd

File: qemu/tests/format_disk.py (hunk 2 above)
Comment (CongSmile) [inline qemu/tests/format_disk.py]: Hi @xutian 

Yes, params["show_dev_cmd"] can save 1 line here.

But I think the way that using params.get() is more easy to understand.

We can talk about it to decide which way is better.

Thanks.

**Comment by ypu** [comment]
Hi @xutian and @CongSmile the useage of params.get() and params[] is different:
- params[] is for the parameters which must exist. If not it will raise an Error and report parameters missing
- params.get() don't request the parameter exist it will return default value or None if it isn't.

So please choose the current way to use them based on your conditions.

**Comment by PyLearner** [comment]
Acked-by : Meng Yang meyang@redhat.com

File: qemu/tests/cfg/format_disk.cfg (hunk 1 above)
Comment (ypu) [inline qemu/tests/cfg/format_disk.cfg]: @CongSmile Double check the code and maybe the code of format disk need some update. So it is OK to still separated the cmds like this. Let's use other pullreqs to fix this problem. And thanks for your explain.

**Comment by CongSmile** [comment]
Hi @xutian 

I have updated the code.
Could you help review it again?

Thanks.

File: qemu/tests/cfg/format_disk.cfg (hunk 1 above)
Comment (CongSmile) [inline qemu/tests/cfg/format_disk.cfg]: Hi @ypu 

Ok, will fix it in other PR.

Thanks for the help.

**Comment by CongSmile** [comment]
Hi xutian,

Have removed those lines.

Could you help review it again?

Thanks.

**Comment by xutian** [comment]
LGTM,

Acked-by: Xu Tian <xutian@redhat.com>

**Comment by xutian** [comment]
thanks @CongSmile  @ypu and  @PyLearner.

Merged.

**Pull request merged**